src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,204 @@
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(LSP.CodeAction request)
+            => ((JToken)request.Data!).ToObject<CodeActionResolveData>()!.TextDocument;
+
+        public async Task<LSP.CodeAction> HandleRequestAsync(LSP.CodeAction codeAction, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.GetRequiredDocument();
+
+            var data = ((JToken)codeAction.Data!).ToObject<CodeActionResolveData>();
+            Assumes.Present(data);
+
+            var options = _globalOptions.GetCodeActionOptionsProvider();
+
+            var codeActionsCache = context.GetRequiredLspService<CodeActionsCache>();
+            var codeActions = await CodeActionHelpers.GetCodeActionsAsync(
+                codeActionsCache,
+                document,
+                data.Range,
+                options,
+                _codeFixService,
+                _codeRefactoringService,
+                cancellationToken).ConfigureAwait(false);
+
+            var codeActionToResolve = CodeActionHelpers.GetCodeActionToResolve(data.UniqueIdentifier, codeActions);
+            Contract.ThrowIfNull(codeActionToResolve);
+
+            var operations = await codeActionToResolve.GetOperationsAsync(cancellationToken).ConfigureAwait(false);
+
+            // We only support making solution changing actions.  We can also ignore DocumentNavigation as that is an
+            // additional suggestion a code action can make about what to do post edit.
+            var applicableActions = operations.Where(o => o is ApplyChangesOperation or DocumentNavigationOperation).ToArray();
+
+            // TO-DO: We currently must execute code actions which add new documents on the server as commands,
+            // since there is no LSP support for adding documents yet. In the future, we should move these actions
+            // to execute on the client.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+
+            var solution = document.Project.Solution;
+            var textDiffService = solution.Services.GetService<IDocumentTextDifferencingService>();
+
+            using var _ = ArrayBuilder<TextDocumentEdit>.GetInstance(out var textDocumentEdits);
+
+            foreach (var option in operations)
+            {
+                // We only support making solution-updating operations in LSP.  And only ones that modify documents. 1st
+                // class code actions that do more than this are supposed to add the CodeAction.MakesNonDocumentChange
+                // in their Tags so we can filter them out before returning them to the client.
+                //
+                // However, we cannot enforce this as 3rd party fixers can still run.  So we filter their results to 
+                // only apply the portions of their work that updates documents, and nothing else.
+                if (option is not ApplyChangesOperation applyChangesOperation)
```

Comment:
lets log an info message here indicating the code action and what operation was skipped.  Will make it easier when someone asks why their code action broke to see what happened